core/src/main/resources/hudson/Messages.properties

```diff
@@ -67,8 +67,8 @@ PluginManager.UnexpectedException=Unexpected exception going through the retryin
 
 PluginManager.compatWarning=\
  Warning: The new version of this plugin is marked as incompatible with the installed version. \
- This is usually the case because its behavior changed, or because it uses a different settings format than the installed version. \
- Jobs using this plugin may need to be reconfigured, and/or you may not be able to cleanly revert to the prior version without manually restoring old settings. \
+ This is usually the case because its behavior or APIs changed, or because it uses a different settings format than the installed version. \
+ Other plugins with a dependency on this plugin may be incompatible with this update and no longer work as expected, jobs using this plugin may need to be reconfigured, and/or you may not be able to cleanly revert to the prior version without manually restoring old settings. \
```

Comment:
this could also affect Jenkinsfiles and pipeline libraries, not sure if this should be explicitly mentioned

Reply:
I think that's covered by

> jobs using this plugin may need to be reconfigured

There's a certain indirection here with libraries, but it's close enough.